src/main/java/game/entities/items/weapons/Weapon.java

```diff
@@ -46,9 +46,20 @@ public Weapon setAttackRange(int attackRange) {
       this.attackRange = attackRange; 
       return this;
     }
+    public Weapon setDurability(int durability) {
+      this.durability = durability;
+      return this;
+    }
 
     // Class getters
     public int getWeaponDamage() { return weaponDamage; }
     public int getAttackSpeed() { return attackSpeed; }
     public int getAttackRange() { return attackRange; }
+    public int getDurability() { return durability; }
+
+    // Method for any weapon to take damage.
+    public void takeDamage() {
+      Damage dmg = new Damage(this);
+      dmg.addDamagetoWeapon(1);
```

Comment:
Remove this and replace with the method that just makes a `GameObject` take damage overall.

Reply:
It does that now asshole